huaweicloud/services/iotda/data_source_huaweicloud_iotda_batchtasks.go

```diff
@@ -0,0 +1,255 @@
+						},
+						"targets_filter": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"task_policy": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"schedule_time": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"retry_count": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"retry_interval": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status_desc": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"task_progress": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"total": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"processing": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"success": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"fail": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"waitting": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"fail_wait_retry": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"stopped": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"removed": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"created_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceBatchTasksRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg       = meta.(*config.Config)
+		region    = cfg.GetRegion(d)
+		isDerived = WithDerivedAuth(cfg, region)
+	)
+
+	client, err := cfg.HcIoTdaV5Client(region, isDerived)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	var (
+		allTasks []model.Task
+		limit    = int32(50)
+		offset   int32
+	)
+
+	for {
+		listOpts := model.ListBatchTasksRequest{
+			AppId:    utils.StringIgnoreEmpty(d.Get("space_id").(string)),
+			TaskType: d.Get("type").(string),
+			Status:   utils.StringIgnoreEmpty(d.Get("status").(string)),
+			Limit:    utils.Int32(limit),
+			Offset:   &offset,
+		}
+
+		listResp, listErr := client.ListBatchTasks(&listOpts)
+		if listErr != nil {
+			return diag.Errorf("error querying IoTDA batch tasks: %s", listErr)
+		}
+
+		if len(*listResp.Batchtasks) == 0 {
```

Comment:
Please determine in advance whether the pointer type is nil.

Reply:
fixed.